tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -767,6 +949,53 @@ private void InitializeAasTags()
             }
         }
 
+        private bool IsAllowedAttributeType(object value)
+        {
+            if (value is null)
+            {
+                return false;
+            }
+
+            if (value is Array array)
+            {
+                if (array.Length == 0 ||
+                    array.Rank > 1)
+                {
+                    // Newtonsoft doesn't seem to support multidimensional arrays (e.g., [,]), but does support jagged (e.g., [][])
+                    return false;
+                }
+
+                if (value.GetType() is { } type
+                 && type.IsArray
+                 && type.GetElementType() == typeof(object))
+                {
+                    // Arrays may only have a primitive type, not 'object'
+                    return false;
+                }
+
+                value = array.GetValue(0);
+
+                if (value is null)
+                {
+                    return false;
+                }
+            }
+
+            return (value is string or bool ||
```

Comment:
To keep our two span event serializations in sync, I think we should drop `decimal` and `ulong` types from being serialized. I think this change alone would help with the difference I noted during the native span events serialization.

Note: This could be considered a breaking change, but I think that we could call it out and note this change is in alignment with the allowed attribute types in OpenTelemetry.

Reply:
Did that one on: https://github.com/DataDog/dd-trace-dotnet/commit/324c4372c1d35f5b23bc9221643fab7d368516ae